src/ledger/insert_shred.zig

```diff
@@ -739,6 +797,8 @@ pub const ShredInserter = struct {
         allocator: std.mem.Allocator,
         slot: Slot,
         working_set: *AutoHashMap(Slot, IndexMetaWorkingSetEntry),
+        // QUESTION: Is there another way apart from passing this down? What about passing self.metrics instead?
+        // See index_meta_time_us was not initially sure why this? But this is being passed all the way down for timing purposes
```

Comment:
You're right. This is messy. How about this: https://github.com/Syndica/sig/commit/c4c0a754adf778812609b47d1f1bf30699185668